### PR TITLE
Add support for unix domain sockets and named pipes

### PR DESCRIPTION
**Pull request by mfussenegger**
For adapters like cmake which don't support stdin or TCP

Closes https://github.com/mfussenegger/nvim-dap/issues/924

For example:

    local dap = require("dap")
    dap.adapters.cmake = {
      type = "pipe",
      pipe = "${pipe}",
      executable = {
        command = "cmake",
        args = {"--debugger", "--debugger-pipe", "${pipe}"}
      }
    }
    dap.configurations.cmake = {
      {
        name = "Build",
        type = "cmake",
        request = "launch",
      }
    }

### CODE REVIEW

File: doc/dap.txt [hunk 1]
```diff
@@ -76,6 +76,9 @@ The `Adapter` needs to contain a `type`, which can be one of:
 - `server`, to indicate that nvim-dap can connect to an already-running
   debug adapter via TCP.
 
+- `pipe`, to connect to an already running debug adapter via a unix domain
```
Comment (FredeEB) [inline doc/dap.txt:79]: Does it have to be already running? Doesn't the executable field in the adapter spec make this invalid?

Comment (mfussenegger) [reply doc/dap.txt]: I re-phrased it a bit to (hopefully) clarify it.

Comment (FredeEB) [reply doc/dap.txt]: Beautiful! Thanks for this :rocket:

**Comment by Willem-J-an** [comment]
Nice! It also works for powershell! I think the powershell dap server has some weird way of interactive input; it wants to use the console as repl, but if I enable that setting it leads to errors. The regular repl does not return any output.
```
local PSES_BUNDLE_PATH = os.getenv("HOME") .. "/.local/share/nvim/mason/packages/powershell-editor-services"
local tmpdir = os.tmpname() .. 'd'
os.execute("mkdir " .. tmpdir)
dap.adapters.ps1 = {
	type = "pipe",
	pipe = "${pipe}",
	executable = {
		command = "pwsh",
		args = {
			"-NoLogo",
			"-NoProfile",
			"-File",
			PSES_BUNDLE_PATH .. "/PowerShellEditorServices/Start-EditorServices.ps1",
			"-BundledModulesPath",
			PSES_BUNDLE_PATH,
			"-LogPath",
			tmpdir .. "/logs.log",
			"-SessionDetailsPath",
			tmpdir .. "/session.json",
			"-FeatureFlags",
			"@()",
			"-AdditionalModules",
			"@()",
			"-HostName",
			"My Client",
			"-HostProfileId",
			"myclient",
			"-HostVersion",
			"1.0.0",
			"-LogLevel",
			"Normal",
			"-DebugServiceOnly",
			"-DebugServicePipeName",
			"${pipe}",
			"-WaitForDebugger"
		},
	},
}

```
```
        {
            "name": "PowerShell: Current",
            "type": "ps1",
            "request": "launch",
            "script": "${file}",
            "cwd": "${file}"
        },
```

**Pull request merged**